frontend/src/metabase/parameters/components/ParameterSidebar/ParameterSidebar.unit.spec.tsx

```diff
@@ -78,4 +88,29 @@ describe("ParameterSidebar", () => {
     labelInput.blur();
     expect(labelInput).toHaveValue("Bar");
   });
+
+  it("if the parameter updates, the label should update (metabase#34611)", () => {
+    const initialParameter = createMockUiParameter({
+      id: "id1",
+      name: "Foo",
+      slug: "foo",
+    });
+    const otherParameter = createMockUiParameter({
+      id: "id1",
+      name: "Baz",
+      slug: "baz",
+    });
+    const { setParameter } = setup({
+      initialParameter,
+      otherParameters: [otherParameter],
+    });
+
+    const labelInput = screen.getByLabelText("Label");
+    expect(labelInput).toHaveValue("Foo");
+
+    act(() => {
```

Comment:
Usually we don't use `act` in our tests since it's generally bad practice with React Testing Library, where functions like RTL's `render` are already wrapped with `act`. Since you're using a test wrapper for the function already, maybe you can add a button that triggers the `setParameter` function, then check for the `labelInput` value?

[Context here](https://flyingsquirrel.medium.com/when-should-i-use-act-in-react-testing-library-d7dd22a3340e#:~:text=You%20don't%20need%20to%20use%20act()%20actually.,-The%20reason%20why&text=testing%2Dlibrary%2Freact%20overrides%20eventWrapper,are%20executed%20in%20act()%20.&text=Therefore%20you%20don't%20have,like%20userEvent%20with%20act()%20.&text=act%20calls%20can%20be%20nested,actScopeDepth%20based%20on%20the%20stack.)

Reply:
This is what I'm thinking in terms of how the test should be structured - let me know what you think

https://github.com/metabase/metabase/commit/917bf2f058a713c5149b82e0d05a3995049d8831